app/src/main/AndroidManifest.xml

```diff
@@ -35,6 +35,8 @@
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
 
+    <uses-permission android:name="android.permission.MANAGE_EXTERNAL_STORAGE"/>
```

Comment:
This permission is a big red flag. 🚩 

You should not need access to the entire storage. You only need access to a single directory to store the logs. In fact, we could even store the logs in an app-private directory and provide an export button, to share it to another app via a content URI.

As a security-conscious user, if an app requests permission to access the entire storage, this will raise some eyebrows. Some apps have a good reason (e.g.  file managers), but a Bluetooth Spam app should not need that.

Ideally, this should be rewritten to use app-local storage or Scoped Storage.